src/server/system_services/bucket_server.js

```diff
@@ -1943,17 +1943,25 @@ async function delete_bucket_replication(req) {
     await replication_store.instance().delete_replication_by_id(replication_id);
 }
 
-function validate_replication(req) {
+async function validate_replication(req) {
     const replication_rules = req.rpc_params.replication_policy.rules;
     // num of rules in configuration must be in defined limits
     if (replication_rules.length > config.BUCKET_REPLICATION_MAX_RULES ||
         replication_rules.length < 1) throw new RpcError('INVALID_REPLICATION_POLICY', 'Number of rules is invalid');
 
     const rule_ids = [];
     const pref_by_dst_bucket = {};
+    const src_bucket = req.system.buckets_by_name && req.system.buckets_by_name[req.name];
```

Comment:
why not get it from the rule? 
At this point we do not care about the bucket, only about it's name

Reply:
Thanks, updated the same.